File: app/src/main/java/com/sagalogistics/backend/models/Order.java
package com.sagalogistics.backend.models;

import java.io.Serializable;
import java.util.Map;

public interface Order extends Serializable {
    String getKey();

    void setKey(String key);

    Map<String, Integer> getItems();

    void setItems(Map<String, Integer> items);

    void updateItem(String key, Integer quantity);

    Integer getQuantityItem(String key);

    void removeItem(String key);
}

File: app/src/main/java/com/sagalogistics/backend/models/Order.kt
package com.sagalogistics.backend.models

import java.io.Serializable

interface Order : Serializable {
    var key: String?
    var items: MutableMap<String, Int>
    fun updateItem(key: String, quantity: Int)
    fun getQuantityItem(key: String): Int?
    fun removeItem(key: String)
}